rllib/core/rl_trainer/torch/torch_rl_trainer.py

```diff
@@ -107,33 +103,44 @@ def build(self) -> None:
         # TODO (Kourosh): Instead of using _TorchAccelerator, we should use the public
         # api in ray.train but allow for session to be None without any errors raised.
         if self._use_gpu:
-            self._device = _TorchAccelerator().get_device()
+            # _TorchAccelerator().get_device() returns the 0th device if
+            # it is called from outside of a Ray Train session. Its necessary to give
+            # the user the option to run on the gpu of their choice, so we enable that
+            # option here via the local gpu id scaling config parameter.
+            if self._distributed:
+                self._device = _TorchAccelerator().get_device()
+            else:
+                assert self._local_gpu_idx < torch.cuda.device_count(), (
+                    f"local_gpu_idx {self._local_gpu_idx} is not a valid GPU id or is "
+                    " not available."
+                )
+                # this is an index into the available cuda devices. For example if
+                # os.environ["CUDA_VISIBLE_DEVICES"] = "1" then
+                # torch.cuda.device_count() = 0 and torch.device(0) will actuall map to
```

Comment:
nit: this should be `torch.cuda.device_count() = 1`, no?

Reply:
ah fixed